lib/screens/cart_screen.dart

```diff
@@ -72,3 +63,49 @@ class CartScreen extends StatelessWidget {
     );
   }
 }
+
+class OrderButton extends StatefulWidget {
+  const OrderButton({
+    Key key,
+    @required this.cart,
+  }) : super(key: key);
+
+  final Cart cart;
+
+  @override
+  _OrderButtonState createState() => _OrderButtonState();
+}
+
+class _OrderButtonState extends State<OrderButton> {
+  var _isLoading = false;
+
+  void _loading() {
+    setState(() {
+      _isLoading = true;
+    });
+  }
+
+  void _notLoading() {
+    setState(() {
+      _isLoading = false;
+    });
+  }
```

Comment:
DRY. just make the `_loading()` method have a parameter where you can change the value of `_isLoading`